include/bpf2c.h

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+#pragma once
+#include <stdint.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include "ebpf_structs.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#define UBPF_STACK_SIZE 512
+
+#define IMMEDIATE(X) (int32_t) X
+#define OFFSET(X) (int16_t) X
+#define POINTER(X) (uint64_t)(X)
+
+#if !defined(htobe16)
+#define htobe16(X) swap16(X)
+#define htobe32(X) swap32(X)
+#define htobe64(X) swap64(X)
+
+#define htole16(X) (X)
+#define htole32(X) (X)
+#define htole64(X) (X)
+#endif
+
+    typedef struct _helper_function_entry
+    {
+        uint64_t (*address)(uint64_t r1, uint64_t r2, uint64_t r3, uint64_t r4, uint64_t r5);
+        uint32_t helper_id;
+        const char* name;
+        bool tail_call;
+    } helper_function_entry_t;
+
+    typedef struct _map_entry
+    {
+        void* address;
+        ebpf_map_definition_in_file_t definition;
+        const char* name;
+    } map_entry_t;
+
+    typedef struct _program_entry
```

Comment:
There is a need to convey quite a bit of metadata between the verifier and the execution context (verifier version, helper function hash, etc). I think it makes more sense to inject a blob that is shared between verifier and execution context. We should discuss this and add it in a future PR.

Reply:
At the minimum, we need to export the program and attach types for each program in the ELF file, which will be required to load and attach those programs. Program and attach types are derived from the section name and will be otherwise lost once the conversion from ELF -> driver happens.

But I think we can add that in a separate PR. Is it possible to file a bug for that and add a TODO in the code here?